src/operator/nn/cast_storage-inl.h

```diff
@@ -57,23 +57,41 @@ inline void CastStorageDnsRspImpl(mshadow::Stream<cpu>* s, const TBlob& dns, NDA
       rsp->CheckAndAllocAuxData(rowsparse::kIdx, mshadow::Shape1(num_rows));
       TBlob row_idx_blob = rsp->aux_data(rowsparse::kIdx);
       RType* row_idx = row_idx_blob.dptr<RType>();
+
+      double start = dmlc::GetTime();
       mxnet_op::Kernel<MarkRspRowIdx, cpu>::Launch(s, num_rows, row_idx,
           dns.dptr<DType>(), num_cols);
+      double elapsed1 = dmlc::GetTime() - start;
+
       index_t nnr = 0;
-      nnr = std::accumulate(row_idx, row_idx+num_rows, nnr);
+      start = dmlc::GetTime();
+      nnr = mxnet::common::ParallelAccumulate(row_idx, num_rows, nnr);
+      //nnr = std::accumulate(row_idx, row_idx+num_rows, nnr);
+      double elapsed2 = dmlc::GetTime() - start;
```

Comment:
Yes, I will push another commit to clean up everything. This is kept here for convenience for now.